framework/src/modules/dpos_v2/module.ts

```diff
@@ -602,20 +607,23 @@ export class DPoSModule extends BaseModule {
 		}
 		const shuffledValidators = shuffleDelegateList(randomSeed1, validators);
 		const bftValidators = [];
+		const generators = [];
 		for (const validatorAddress of shuffledValidators) {
 			const validatorAccount = await this._validatorsAPI.getValidatorAccount(
 				apiContext,
 				validatorAddress,
 			);
-			const bftWeight =
-				standbyDelegates.findIndex(addr => addr.equals(validatorAddress)) > -1
-					? BigInt(0)
-					: BigInt(1);
-			bftValidators.push({
+			// if validator is active
+			if (standbyDelegates.findIndex(addr => addr.equals(validatorAddress)) === -1) {
```

Comment:
Would it maybe be better here to check if the delegate is active (instead of not standby). On Mainnet right now there are less standby delegates than active, but in general the set of standby delegates could be `>1000` whereas there are always only 101 active delegates.